matrix/plugins/glitch/actions.py

```diff
@@ -26,14 +29,16 @@ def decorate(self, func):
         object in that set.
 
         """
-        @enforce.runtime_validation
+
         @wraps(func)
-        async def wrapped(model, objects, **kwargs):
+        async def wrapped(rule, model, objects, **kwargs):
             for obj in objects:
-                await func(model, obj, **kwargs)
-        self[func.__name__] = wrapped
+                await enforce.runtime_validation(func(
+                    rule, model, obj, **kwargs))
+        self[func.__name__] = {'func': wrapped, 'args': func.__annotations__}
```

Comment:
There are a number of ways this might be better to do. I am still quite wary of just matching names like we do here (and like I might do a few other places).

I'd suggest (even though it will have the same outcome) that we future proof this a little bit by using inspect.signature. The Signature object it returns should have all the function information we want.